src/animalShelter/AnimalShelter.ts

```diff
@@ -68,8 +69,33 @@ export default class AnimalShelter {
       console.log(
         `Cannot change reptile limit to ${newReptileLimit} there ${reptilesTotal == 1 ? "is" : "are"} already ${reptilesTotal} ${reptilesTotal == 1 ? "reptile" : "reptile"}`
       );
+    };
+  };
+
+  private hasMethod(animal:Animal, methodName:string): boolean{
+    return typeof (animal as any)[methodName] === "function";
+  };
+
+  public report(){
+    let animalsWhoJump = 0;
+    let animalsWhoRun = 0;
+    let anaimalsWhoClimbTrees =0
+    for(const animal of this.animals){
+      if(this.hasMethod(animal, "jump")){
+        animalsWhoJump++
+      }
+      if(this.hasMethod(animal, "run")){
+        animalsWhoRun++;
+      } 
+      if(animal._canClimbTrees){
+        anaimalsWhoClimbTrees++;
+      }
```

Comment:
either use the hasMethod for every activity or add properties like the canClimbTrees, using both ways is inconsistent. also shouldn't the animals who can climb trees actually have a method `climb`?

Reply:
We only have the knowledge if it climbs or not, that is why this is a boolean property and the others are mthods with activities